gpt_bpe.go

```diff
@@ -160,6 +168,20 @@ func NewEncoder(vocabId string) (*GPTEncoder, error) {
 			}
 		}
 	}
+
+	tokenizerSpecialConfig := resources.TokenizerSpecialsConfig{
+		Add_bos_token: false,
+		Add_eos_token: false,
+		Pad_token:     "",
+	}
+	if special, ok := (rsrcs)["tokenizer_config.json"]; ok {
+		if special.Data != nil {
+			if json.Unmarshal(*special.Data, &tokenizerSpecialConfig) != nil {
+				log.Fatal("Error unmarshalling tokenizer_config.json")
+			}
```

Comment:
May be more helpful to write as

```
if special, ok := (rsrcs)["tokenizer_config.json"]; ok {
  if special.Data != nil {
          err := json.Unmarshal(*special.Data, &tokenizerSpecialConfig)
	  if err != nil {
		  log.Fatal("Error unmarshalling tokenizer_config.json: ", err)
	  }
```

As it allows you to see what the actual error was in debugging.